src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -1672,6 +1673,13 @@ def align(
         elif how == "left":
             filter_expression = filter_expression & left_row_pos.is_not_null()
             select_list = result_projected_column_snowflake_quoted_identifiers
+        elif how == "inner":
+            filter_expression = (
+                filter_expression
+                & left_row_pos.is_not_null()
+                & right_row_pos.is_not_null()
+            )
+            select_list = result_projected_column_snowflake_quoted_identifiers
         else:  # outer
```

Comment:
It seems that `how == "right"` is handled in this branch - the same way as `how == "outer"`. Can we have a separate branch for it, and also have a test case that would differentiate between the output of a right align and an outer align? (In other words, a test case where a right align would fail now :))

Reply:
Btw, if you prefer to leave this for a separate PR, I'm fine with that too.

Reply:
@sfc-gh-helmeleegy Yes that's a good point! I updated ordered_dataframe.align to branch out `how==outer` and threw an Error for any other how argument (such as `how=right`) I've also added a test for `how=inner` in test_ordered_dataframe and a negative test for `how=right`. Is this what you were thinking of?

Reply:
I think `how=right` should be an easy followup PR. This one looks good for now.